### PR TITLE
[WIP] Widget/View system

### PR DESCRIPTION
**Pull request by mfalesni**

### CODE REVIEW

**Comment by dajoRH** [comment]
Lint report for commit a684e377ab9026ec859bf31d94660590c656c166:

`selenium_view.py`:
- :grey_exclamation: Line 107:9: **T001** *print statement found.*
- :red_circle: Line 353:9: **N802** *function name should be lowercase*

Please, rectify these issues :smirk: .
*CFME QE Bot*

File: selenium_view.py [hunk 1]
```diff
@@ -0,0 +1,634 @@
...
+        except IndexError:
+            raise NoSuchElementException('Could not find an element {}'.format(repr(locator)))
+
+    def perform_click(self):
+        ActionChains(self.selenium).click().perform()
+
+    def click(self, *args, **kwargs):
+        self.move_to_element(*args, **kwargs)
+        # and then click on current mouse position
+        self.perform_click()
```
Comment (psav) [inline selenium_view.py]: Do we need to ensure page safe before clicking?

Comment (mfalesni) [inline selenium_view.py]: move_to_element ensures that.

**Comment by dajoRH** [comment]
Lint report for commit f1fb3a047c1559815024348659ced76013e03a70:

Everything seems all right :smile: .
*CFME QE Bot*

File: selenium_view.py (hunk 1 above)
Comment (mfalesni) [inline selenium_view.py]: Actually I sticked the call to the "ensurer" in the element query function so :)

**Comment by dajoRH** [comment]
Lint report for commit ed6ae02f3d6408a11fb01a8b4395b03b7ee66d5a:

Everything seems all right :smile: .
*CFME QE Bot*

**Comment by dajoRH** [comment]
Lint report for commit 1ef49190a51d153fe913d3d9bd17ffcdaf6fbf48:

Everything seems all right :smile: .
*CFME QE Bot*

**Comment by dajoRH** [comment]
Would you mind rebasing this Pull Request against latest master, please? :trollface:
*CFME QE Bot*

**Comment by dajoRH** [comment]
Lint report for commit f7fb3917d252f21f7ac900798d0a09b40c5cb7ea:

Everything seems all right :smile: .
*CFME QE Bot*

**Comment by dajoRH** [comment]
Lint report for commit 15f4483797f6ede2d60bf7607d9718587f381532:

Everything seems all right :smile: .
*CFME QE Bot*

**Comment by dajoRH** [comment]
Lint report for commit 8ea98ba239bcb51372fa68b78776bbf4fc6bfb8b:

`cfme/login.py`:
- :red_circle: Line 11:1: **F401** *'ElementNotVisibleException' imported but unused*
- :red_circle: Line 11:1: **F401** *'NoSuchElementException' imported but unused*
- :red_circle: Line 15:1: **F401** *'dashboard' imported but unused*
- :red_circle: Line 17:1: **F401** *'Form' imported but unused*
- :red_circle: Line 17:1: **F401** *'fill' imported but unused*

`cfme/tests/test_login.py`:
- :red_circle: Line 2:1: **F401** *'dashboard' imported but unused*

Please, rectify these issues :smirk: .
*CFME QE Bot*

**Comment by dajoRH** [comment]
Lint report for commit be55349cea9639c75543afc4db37bed8f7b1c376:

`cfme/login.py`:
- :red_circle: Line 12:1: **F401** *'ElementNotVisibleException' imported but unused*
- :red_circle: Line 12:1: **F401** *'NoSuchElementException' imported but unused*
- :red_circle: Line 16:1: **F401** *'dashboard' imported but unused*
- :red_circle: Line 18:1: **F401** *'Form' imported but unused*
- :red_circle: Line 18:1: **F401** *'fill' imported but unused*

`cfme/tests/test_login.py`:
- :red_circle: Line 2:1: **F401** *'dashboard' imported but unused*

Please, rectify these issues :smirk: .
*CFME QE Bot*

**Comment by dajoRH** [comment]
Lint report for commit d421bff174414cc220f1493ae443e28ebfe3d97f:

`cfme/login.py`:
- :red_circle: Line 12:1: **F401** *'ElementNotVisibleException' imported but unused*
- :red_circle: Line 12:1: **F401** *'NoSuchElementException' imported but unused*
- :red_circle: Line 16:1: **F401** *'dashboard' imported but unused*
- :red_circle: Line 18:1: **F401** *'Form' imported but unused*
- :red_circle: Line 18:1: **F401** *'fill' imported but unused*

`cfme/tests/test_login.py`:
- :red_circle: Line 2:1: **F401** *'dashboard' imported but unused*

Please, rectify these issues :smirk: .
*CFME QE Bot*

**Comment by dajoRH** [comment]
Lint report for commit 5a9db3574baa15e1313b5cb29c33809e52f381aa:

`cfme/control/explorer.py`:
- :red_circle: Line 26:1: **F401** *'View' imported but unused*

`cfme/login.py`:
- :red_circle: Line 12:1: **F401** *'ElementNotVisibleException' imported but unused*
- :red_circle: Line 12:1: **F401** *'NoSuchElementException' imported but unused*
- :red_circle: Line 16:1: **F401** *'dashboard' imported but unused*
- :red_circle: Line 18:1: **F401** *'Form' imported but unused*
- :red_circle: Line 18:1: **F401** *'fill' imported but unused*

`cfme/tests/test_login.py`:
- :red_circle: Line 2:1: **F401** *'dashboard' imported but unused*

`smartobj.py`:
- :red_circle: Line 6:7: **N801** *class names should use CapWords convention*

Please, rectify these issues :smirk: .
*CFME QE Bot*

**Comment by dajoRH** [comment]
Lint report for commit 7eae8d2a568c08c7d8a5add15ea32f74b084b6b3:

`cfme/control/explorer.py`:
- :red_circle: Line 26:1: **F401** *'View' imported but unused*

`cfme/login.py`:
- :red_circle: Line 12:1: **F401** *'ElementNotVisibleException' imported but unused*
- :red_circle: Line 12:1: **F401** *'NoSuchElementException' imported but unused*
- :red_circle: Line 16:1: **F401** *'dashboard' imported but unused*
- :red_circle: Line 18:1: **F401** *'Form' imported but unused*
- :red_circle: Line 18:1: **F401** *'fill' imported but unused*

`cfme/tests/test_login.py`:
- :red_circle: Line 2:1: **F401** *'dashboard' imported but unused*

`smartobj.py`:
- :red_circle: Line 6:7: **N801** *class names should use CapWords convention*

Please, rectify these issues :smirk: .
*CFME QE Bot*

**Comment by dajoRH** [comment]
Lint report for commit 36051a8a041a803230c53c0e03c057bbc057ab09:

Everything seems all right :smile: .
*CFME QE Bot*

**Comment by dajoRH** [comment]
Lint report for commit dff8b83ad2db78608daa48150a6457a6d75aabdf:

Everything seems all right :smile: .
*CFME QE Bot*

**Comment by dajoRH** [comment]
Lint report for commit 296767742bae6d6eecf5ff305828628cae2a783e:

`cfme/login.py`:
- :red_circle: Line 12:1: **F401** *'ElementNotVisibleException' imported but unused*
- :red_circle: Line 12:1: **F401** *'NoSuchElementException' imported but unused*
- :red_circle: Line 16:1: **F401** *'dashboard' imported but unused*
- :red_circle: Line 18:1: **F401** *'Form' imported but unused*
- :red_circle: Line 18:1: **F401** *'fill' imported but unused*

`cfme/tests/test_login.py`:
- :red_circle: Line 2:1: **F401** *'dashboard' imported but unused*

Please, rectify these issues :smirk: .
*CFME QE Bot*

File: fixtures/browser.py [hunk 2]
```diff
@@ -106,6 +107,11 @@ def browser():
     return utils.browser.browser
 
 
+@pytest.fixture(scope='function')
+def create_view():
+    return lambda *args, **kwargs: current_appliance.browser.instantiate_view(*args, **kwargs)
```
Comment (RonnyPfannschmidt) [inline fixtures/browser.py]: i propose using utils.appliance.get_or_create_appliance here to make the point of the usage more clear

**Comment by dajoRH** [comment]
Lint report for commit 7bfe89e0ff309d14ad02b360616d67dd919366d0:

Everything seems all right :smile: .
*CFME QE Bot*

**Comment by RonnyPfannschmidt** [comment]
given the usage patters it might make sense to provide a global helper named `open_view` and unifying short usage and the fixture

**Pull request closed without merging**